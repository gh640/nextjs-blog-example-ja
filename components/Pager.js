import Link from "next/link"

const Pager = (props) => {
  const { total, page, perPage, href, asCallback } = props

  const prevPage = page > 1 ? page - 1 : null
  let nextPage = null
  if (page < Math.ceil(total / perPage)) {
    nextPage = page + 1
  }

  return (
    <div className="pager">
      <span className="pager-item">
        {prevPage ? (
          <Link href={href} as={asCallback(prevPage)}>
            <a>{prevPage}</a>
          </Link>
        ) : ``}
      </span>
      <span className="pager-item">{page}</span>
      <span className="pager-item">
        {nextPage ? (
          <Link href={href} as={asCallback(nextPage)}>
            <a>{nextPage}</a>
          </Link>
        ) : ``}
      </span>

      <style jsx>{`
        .pager {
          display: flex;
          flex-direction: row;
          justify-content: center;
          flew-wrap: nowrap;
        }

        .pager-item {
          margin: 0 1em;
        }
      `}</style>
    </div>
  )
}

export default Pager
