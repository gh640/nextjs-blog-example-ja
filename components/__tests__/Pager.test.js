import renderer from "react-test-renderer"

import Pager from "../Pager.js"

jest.mock("next/link", () => {
	return ({ children, ...attrs }) => {
		return <a {...attrs}>{children}</a>
	}
})

describe("Pager", () => {
	it("renders collectly with first page", () => {
		const tree = renderer
			.create(<Pager 
				total={105}
				page={1}
				perPage={10}
				href="/archive/[page]"
				asCallback={(page) => `/archive/${page}`}
			/>)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it("renders collectly with second page", () => {
		const tree = renderer
			.create(<Pager 
				total={105}
				page={2}
				perPage={10}
				href="/archive/[page]"
				asCallback={(page) => `/archive/${page}`}
			/>)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it("renders collectly with last page", () => {
		const tree = renderer
			.create(<Pager 
				total={27}
				page={6}
				perPage={5}
				href="/archive/[page]"
				asCallback={(page) => `/archive/${page}`}
			/>)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
