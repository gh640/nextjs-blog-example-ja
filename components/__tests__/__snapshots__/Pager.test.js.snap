// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pager renders collectly with first page 1`] = `
<div
  className="jsx-2002780001 pager"
>
  <span
    className="jsx-2002780001 pager-item"
  >
    
  </span>
  <span
    className="jsx-2002780001 pager-item"
  >
    1
  </span>
  <span
    className="jsx-2002780001 pager-item"
  >
    <a
      as="/archive/2"
      href="/archive/[page]"
    >
      <a
        className="jsx-2002780001"
      >
        2
      </a>
    </a>
  </span>
</div>
`;

exports[`Pager renders collectly with last page 1`] = `
<div
  className="jsx-2002780001 pager"
>
  <span
    className="jsx-2002780001 pager-item"
  >
    <a
      as="/archive/5"
      href="/archive/[page]"
    >
      <a
        className="jsx-2002780001"
      >
        5
      </a>
    </a>
  </span>
  <span
    className="jsx-2002780001 pager-item"
  >
    6
  </span>
  <span
    className="jsx-2002780001 pager-item"
  >
    
  </span>
</div>
`;

exports[`Pager renders collectly with second page 1`] = `
<div
  className="jsx-2002780001 pager"
>
  <span
    className="jsx-2002780001 pager-item"
  >
    <a
      as="/archive/1"
      href="/archive/[page]"
    >
      <a
        className="jsx-2002780001"
      >
        1
      </a>
    </a>
  </span>
  <span
    className="jsx-2002780001 pager-item"
  >
    2
  </span>
  <span
    className="jsx-2002780001 pager-item"
  >
    <a
      as="/archive/3"
      href="/archive/[page]"
    >
      <a
        className="jsx-2002780001"
      >
        3
      </a>
    </a>
  </span>
</div>
`;
